Fix flake in MultiVA tests

Seen [here](https://github.com/letsencrypt/boulder/runs/6429733182?check_suite_focus=true):
```
2022/05/13 21:09:39 http: TLS handshake error from 127.0.0.1:41470: tls: client requested unsupported application protocols ([acme-tls/1])
2022/05/13 21:09:39 http: TLS handshake error from 127.0.0.1:44858: Failing on purpose
2022/05/13 21:09:39 http: TLS handshake error from 127.0.0.1:44998: Failing on purpose
2022/05/13 21:09:39 http: TLS handshake error from 127.0.0.1:56036: tls: client requested unsupported application protocols ([acme-tls/1])
2022/05/13 21:09:39 http: TLS handshake error from 127.0.0.1:53058: tls: client offered only unsupported versions: [304 303]
--- FAIL: TestMultiVA (0.04s)
    --- FAIL: TestMultiVA/Local_VA_ok,_remote_VA_internal_err,_no_enforce_multi_VA (0.00s)
        va_test.go:515: 0 != 1
FAIL
FAIL	github.com/letsencrypt/boulder/va	1.858s
```

Well it turns out that if you just run the Multi VA tests ten thousand times, you get a variety of different flaky failures:
```
❯ docker-compose run boulder go test -run=TestMultiVA -count=10000 ./va/...  
<snip>
--- FAIL: TestMultiVAEarlyReturn (10.01s)
    --- FAIL: TestMultiVAEarlyReturn/One_slow_remote_VA,_early_return (10.01s)
        va_test.go:603: Expected an early return from PerformValidation in < 500 ms, took 10009 ms
--- FAIL: TestMultiVA (67.14s)
    --- FAIL: TestMultiVA/Local_and_remote_VAs_OK,_enforce_multi_VA (10.00s)
        va_test.go:509: expected no prob, got problemType:"connection" detail:"127.0.0.1: Fetching http://localhost/.well-known/acme-challenge/LoqXcYV8q5ONbJQxbmR7SCTNo3tiAXDfowyjxAjEuX0: Timeout during connect (likely firewall problem)" httpStatus:400
    --- FAIL: TestMultiVA/Local_and_remote_VAs_OK,_no_enforce_multi_VA (10.01s)
        va_test.go:509: expected no prob, got problemType:"connection" detail:"127.0.0.1: Fetching http://localhost/.well-known/acme-challenge/LoqXcYV8q5ONbJQxbmR7SCTNo3tiAXDfowyjxAjEuX0: Timeout during connect (likely firewall problem)" httpStatus:400
    --- FAIL: TestMultiVA/Local_VA_bad,_remote_VAs_OK,_no_enforce_multi_VA (10.01s)
        va_test.go:512: "connection" != "unauthorized"
    --- FAIL: TestMultiVA/Local_VA_bad,_remote_VAs_OK,_enforce_multi_VA (10.01s)
        va_test.go:512: "connection" != "unauthorized"
    --- FAIL: TestMultiVA/Local_VA_ok,_remote_VA_internal_err,_enforce_multi_VA (10.01s)
        va_test.go:512: "connection" != "serverInternal"
    --- FAIL: TestMultiVA/Local_VA_ok,_remote_VA_internal_err,_no_enforce_multi_VA (10.01s)
        va_test.go:509: expected no prob, got problemType:"connection" detail:"127.0.0.1: Fetching http://localhost/.well-known/acme-challenge/LoqXcYV8q5ONbJQxbmR7SCTNo3tiAXDfowyjxAjEuX0: Timeout during connect (likely firewall problem)" httpStatus:400
--- FAIL: TestMultiVAEarlyReturn (1.02s)
    --- FAIL: TestMultiVAEarlyReturn/One_slow_remote_VA,_early_return (1.02s)
        va_test.go:603: Expected an early return from PerformValidation in < 500 ms, took 1021 ms
FAIL
FAIL    github.com/letsencrypt/boulder/va       154.653s
?       github.com/letsencrypt/boulder/va/proto [no test files]
FAIL
```

> Basically the root cause here is that the VA tests spin up a miniature HTTP server, bind it to a port on localhost, and use that to handle the outgoing requests from the VA to perform validation. In turn, this setup is necessary because the VA doesn't have an abstraction over a HTTP client that could be mocked out during tests: it [builds an http.Client from scratch](https://github.com/letsencrypt/boulder/blob/main/va/http.go#L595).

IMO since the work of the VA unittests is so fundamentally about performing HTTP requests, I think it's useful for it to actually perform the HTTP requests. Or another way of thinking about it: VA's correctness is deeply intertwined with specific behaviors of Go's `http.Client`. If we cut `http.Client` out of the loop we will miss important bugs.

The specific flakes you show in https://github.com/letsencrypt/boulder/issues/6119#issuecomment-1132279105 are of the variety "we expected X to be fast, but it was slow." Those are really likely to be flaky because they rely on timing. If the box running the test was slow, the test will flake. We should remove or rewrite those tests (or if we decide to keep them, give them a much more generous timing).

The flake from the top of this issue seems different:

```
--- FAIL: TestMultiVA (0.04s)
    --- FAIL: TestMultiVA/Local_VA_ok,_remote_VA_internal_err,_no_enforce_multi_VA (0.00s)
        va_test.go:515: 0 != 1
FAIL
FAIL	github.com/letsencrypt/boulder/va	1.858s
```

The test code the flake hits is:

```
			if tc.ExpectedLog != "" {
				lines := mockLog.GetAllMatching(tc.ExpectedLog)
				test.AssertEquals(t, len(lines), 1)
			}
```

(We should fix this test code so it tells us what line was expected and what lines were emitted). The log line looked for was:

```
	expectedInternalErrLine := fmt.Sprintf(
		`ERR: \[AUDIT\] Remote VA "broken".PerformValidation failed: %s`,
		errBrokenRemoteVA.Error())
```

So I'm guessing this is something like logging the wrong thing. Probably our next step should be to improve the output of this test when it fails.

I believe the root cause is the same. This test case is "no enforce multi VA", which means that the remote results are processed in a goroutine that the main validation process [doesn't wait on](https://github.com/letsencrypt/boulder/blob/50f6a6b84fac218204dff4bff58a209145f05d23/va/va.go#L673-L675). If the remote validation decides to be slow, then it won't have returned, and nothing will have been logged, by the time the main validation routine completes and the test checks the logs.

This can be seen in improved logs:
```
--- FAIL: TestMultiVA (0.01s)
    --- FAIL: TestMultiVA/Local_VA_ok,_remote_VA_internal_err,_no_enforce_multi_VA (0.00s)
        va_test.go:518: Got log [DEBUG: Resolved addresses for localhost: [127.0.0.1] INFO: [AUDIT] Checked CAA records for localhost, [Present: false, Account ID: unknown, Challenge: http-01, Valid for issuance: true] Response="" INFO: [AUDIT] Attempting to validate HTTP-01 for "localhost" with GET to "http://localhost/.well-known/acme-challenge/LoqXcYV8q5ONbJQxbmR7SCTNo3tiAXDfowyjxAjEuX0" INFO: [AUDIT] Validation result JSON={"Hostname":"localhost","Challenge":{"type":"http-01","status":"pending","token":"LoqXcYV8q5ONbJQxbmR7SCTNo3tiAXDfowyjxAjEuX0","keyAuthorization":"LoqXcYV8q5ONbJQxbmR7SCTNo3tiAXDfowyjxAjEuX0.9jg46WB3rR_AHD-EBXdN7cBkH1WOu0tA3M9fm21mqTI","validationRecord":[{"url":"http://localhost/.well-known/acme-challenge/LoqXcYV8q5ONbJQxbmR7SCTNo3tiAXDfowyjxAjEuX0","hostname":"localhost","port":"34847","addressesResolved":["127.0.0.1"],"addressUsed":"127.0.0.1"}]},"ValidationLatency":1653003658.113}]; expected "ERR: \\[AUDIT\\] Remote VA \"broken\".PerformValidation failed: brokenRemoteVA is broken"
```

This shows that the main VA routine was perfectly happy with its own validation, but didn't get and log any remote validation results before finishing.

--- a/va/va_test.go
+++ b/va/va_test.go
@@ -514,7 +514,9 @@ func TestMultiVA(t *testing.T) {
 
 			if tc.ExpectedLog != "" {
 				lines := mockLog.GetAllMatching(tc.ExpectedLog)
-				test.AssertEquals(t, len(lines), 1)
+				if len(lines) != 1 {
+					t.Fatalf("Got log %v; expected %q", mockLog.GetAll(), tc.ExpectedLog)
+				}
 			}
 		})
 	}

--- a/va/va_test.go
+++ b/va/va_test.go
@@ -420,8 +420,9 @@ func TestMultiVA(t *testing.T) {
 			},
 			AllowedUAs: allowedUAs,
 			Features:   noEnforceMultiVA,
-			// The real failure cause should be logged
-			ExpectedLog: expectedInternalErrLine,
+			// Like above, the real failure cause will be logged eventually, but that
+			// will happen asynchronously. It's not guaranteed to happen before the
+			// test case exits, so we don't check for it here.
 		},
 		{
 			// With only one working remote VA there should *not* be a validation

